src/OrchardCore/OrchardCore.DisplayManagement/Shapes/DateTimeShapes.cs

```diff
@@ -122,58 +131,111 @@ public IHtmlContent Duration(TimeSpan? timeSpan)
             return HtmlString.Empty;
         }
 
-        var days = (int)timeSpan.Value.TotalDays;
-        var hours = timeSpan.Value.Hours;
-        var minutes = timeSpan.Value.Minutes;
-        var seconds = timeSpan.Value.Seconds;
+        var tag = new TagBuilder("span");
+
+        tag.AddCssClass("timespan-preview-value");
+
+        tag.Attributes["title"] = timeSpan.ToString();
+
+        tag.InnerHtml.AppendHtml(GetDuration(timeSpan.Value));
+
+        return tag;
+    }
 
-        var builder = new HtmlContentBuilder();
+    private LocalizedHtmlString GetDuration(TimeSpan timeSpan)
+    {
+        var days = timeSpan.Days;
+        var hours = timeSpan.Hours;
+        var minutes = timeSpan.Minutes;
+        var seconds = timeSpan.Seconds;
 
         if (days > 0)
         {
-            builder.AppendHtml(H.Plural(days, "{1} day", "{1} days", days));
+            return GetDurationInDays(timeSpan, days, hours, minutes, seconds);
         }
 
         if (hours > 0)
         {
-            if (builder.Count > 0)
-            {
-                builder.AppendWhitespace();
-            }
-            builder.AppendHtml(H.Plural(hours, "{1} hour", "{1} hours", hours));
+            return GetDurationInHours(timeSpan, hours, minutes, seconds);
         }
 
         if (minutes > 0)
         {
-            if (builder.Count > 0)
-            {
-                builder.AppendWhitespace();
-            }
+            return GetDurationInMinutes(timeSpan, minutes, seconds);
+        }
+
+        return H.Plural(seconds, "One second", "{0} seconds");
+    }
 
-            builder.AppendHtml(H.Plural(hours, "{1} minute", "{1} minutes", minutes));
+    private LocalizedHtmlString GetDurationInMinutes(TimeSpan timeSpan, int minutes, int seconds)
+    {
+        if (seconds == timeSpan.TotalSeconds)
```

Comment:
If the timespan is only seconds (this test) then why not display `"1 second, "{0} seconds"` ? I think you made a mistake here because I see hours is handled correctly.

Reply:
fixed